b2/b2http.py

```diff
@@ -0,0 +1,178 @@
+######################################################################
+#
+# File: b2/b2http.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from __future__ import print_function
+
+import json
+import requests
+import six
+import socket
+
+
+class WebApiError(Exception):
+    """
+    Holds information extracted from a call to a B2 API.
+
+    When an error happens, B2 returns an error JSON that looks like this:
+
+        {
+            "status" : 400,
+            "code" : "invalid_bucket_name",
+            "message" : "bucket name is too long"
+        }
+    """
+
+    def __init__(self, status, code, message):
+        super(WebApiError, self).__init__(status, code, message)
+        self.status = status
+        self.code = code
+        self.message = message
+
+    def __repr__(self):
+        return "WebApiError(%d, '%s', '%s')" % (self.status, self.code, self.message)
+
+    def __eq__(self, other):
+        return repr(self) == repr(other)
```

Comment:
This seems weird, exceptions don't usually have this. It is needed for tests, but checking `code` field is enough there. In test we want to check behavior, not identity.